lib/feature/add_seed_to_app/enter_seed_phrase/view/enter_seed_phrase_view.dart

```diff
@@ -0,0 +1,414 @@
+                    ),
+                  ],
+                ),
+                const SizedBox(height: 12),
+                Text(
+                  l10n.paste_seed_into_first_box,
+                  style: StyleRes.bodyText.copyWith(color: colors.textPrimary),
+                ),
+                const SizedBox(height: 24),
+                Row(
+                  children: [
+                    Expanded(
+                      child: Column(
+                        children: activeControllers
+                            .getRange(0, value ~/ 2)
+                            .mapIndexed(
+                              (index, c) =>
+                                  _inputBuild(c, focuses[index], index + 1),
+                            )
+                            .toList(),
+                      ),
+                    ),
+                    const SizedBox(width: 16),
+                    Expanded(
+                      child: Column(
+                        children: activeControllers
+                            .getRange(value ~/ 2, value)
+                            .mapIndexed(
+                          (index, c) {
+                            final i = index + value ~/ 2;
+                            return _inputBuild(c, focuses[i], i + 1);
+                          },
+                        ).toList(),
+                      ),
+                    ),
+                  ],
+                ),
+                ValueListenableBuilder<String?>(
+                  valueListenable: formErrorNotifier,
+                  builder: (context, error, __) {
+                    return Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        const SizedBox(height: 8),
+                        if (error == null)
+                          SizedBox(
+                            height: StyleRes.regular16.fontSize! *
+                                StyleRes.regular16.height!,
+                          )
+                        else
+                          Text(
+                            error.capitalize!,
+                            style: StyleRes.regular16
+                                .copyWith(color: colors.textNegative),
+                          ),
+                        const SizedBox(height: 16),
+                      ],
+                    );
+                  },
+                ),
+              ],
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  /// [index] start with 1
+  Widget _inputBuild(
+    TextEditingController controller,
+    FocusNode focus,
+    int index,
+  ) {
+    return Builder(
+      builder: (context) {
+        final colors = context.themeStyle.colors;
+        return Padding(
+          padding: const EdgeInsets.only(bottom: 8),
+          child: CommonInput(
+            controller: controller,
+            suggestionsCallback: (_) => _suggestionsCallback(controller),
```

Comment:
I think all business logic should be done via bloc. Because the interface code should be so stupid so we can mock ALL it's state using bloc state. Adding business logic in UI creates complications and prevents us to write tests at all.

Reply:
-